modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2032,63 +2032,32 @@ def map_fn(df):
             new_dtypes,
         )
 
-    @lazy_metadata_decorator(apply_axis="both")
-    def sort_by(
-        self,
-        axis: Union[int, Axis],
-        columns: Union[str, List[str]],
-        ascending: bool = True,
-        **kwargs,
-    ) -> "PandasDataframe":
+    def _apply_func_to_range_partitioning(
+        self, key_column, func, ascending=True, **kwargs
+    ):
```

Comment:
moved the logic of `sort_by` preceding the actual reshuffling to a separate method called `_apply_func_to_range_partitioning`, the logic was mostly copied untouched